[IMP] marketing: sms essentials add block server action (#8820)
odoo/documentation: content/applications/marketing/sms_marketing/essentials/sms_essentials.rst

==============

Utilizing :abbr:`SMS (Short Message Service)` outreach in communication strategies can help

> companies expand their market reach, especially in some countries, where emails might not be common,
> or even used at all.

Reviewer: ```suggestion
companies expand their market reach, especially in countries where emails might not be common,
or even used at all.
```

   Odoo displays all previously created mailing lists, along with various types of data related to
   that specific list (e.g. number of contacts, mailings, recipients, etc.).

>    To learn more about mailing lists and contacts, check out :doc:`mailing_lists_blacklists`

Reviewer: ```suggestion
   To learn more about mailing lists and contacts, check out :doc:`mailing_lists_blacklists`.
```

.. image:: sms_essentials/sms-mailing-list.png
   :align: center
   :alt: View of the mailing list page in the SMS marketing application.

To reveal all the possible options in the :guilabel:`Recipients` field, click the field to see all
the choices Odoo makes available.

When another field (other than :guilabel:`Mailing List`) is selected, the option to specify that
chosen field even further becomes available — either with a default recipient filter equation that
appears automatically (which can be customized to fit any business need), or, if no default
recipient filter equation is present, an :guilabel:`Add Filter` button will appear.

> Clicking the :guilabel:`Add Filter` button, reveals fully customizable domain rule fields, which can

Reviewer: ```suggestion
Clicking the :guilabel:`Add Filter` button reveals fully customizable domain rule fields, which can
```

   :align: center
   :alt: Reporting page in SMS Marketing.

.. tip::
   SMS messages can be sent using server actions in Odoo. To do so, in :ref:`developer mode
   <developer-mode>`, navigate to :menuselection:`Settings app --> Technical menu --> Automation -->

>    Automated Actions`. Click :guilabel:`Create` or :guilabel:`New` to create a new action.

Reviewer 2: Since this is aimed at 15.0, it should just say "Create," right?
```suggestion
   Automated Actions`. Then, click :guilabel:`Create` to create a new action.
```

>    Enter an :guilabel:`Action Name` and select a :guilabel:`Model` to implement this action on. Set

Reviewer 2: ```suggestion
   Enter an :guilabel:`Action Name`, and select a :guilabel:`Model` to implement this action on. Set
```
do you think it's important to reference/explain the 'Active' toggle switch that appears between 'Model' and 'Trigger'?

>    Under :guilabel:`Trigger` set either :guilabel:`On Creation`, :guilabel:`On Update`,

Reviewer: ```suggestion
   Set the :guilabel:`Trigger` to either :guilabel:`On Creation`, :guilabel:`On Update`,
```

   :guilabel:`On Creation & Update`, :guilabel:`On Deletion`, :guilabel:`Based on Form

>    Modification`, :guilabel:`Based on Timed Condition`.

Reviewer: ```suggestion
   Modification`, or :guilabel:`Based on Timed Condition`.
```

   Based on the selection for the :guilabel:`Trigger`, additional fields may populate. Ensure a
   model has been selected before creating any record filters.

> 

Reviewer 2: Is there a reason the 'Apply On' field wasn't mentioned here?

>    Under :guilabel:`Action To Do`, select :guilabel:`Send SMS Text Message`. Next, set the

Reviewer 2: ```suggestion
   Under :guilabel:`Action To Do` drop-down field, select :guilabel:`Send SMS Text Message`. Next, set the
```

>    Under :guilabel:`Action To Do` select :guilabel:`Send SMS Text Message`. Next, set the
>    :guilabel:`SMS Template` and choose whether the SMS message should be logged as a note, by

Reviewer: ```suggestion
   Under :guilabel:`Action To Do`, select :guilabel:`Send SMS Text Message`. Next, set the
   :guilabel:`SMS Template`, and choose whether the SMS message should be logged as a note, by
```

>    ticking the box next to :guilabel:`Log as note`.

Reviewer 2: ```suggestion
   ticking the box next to :guilabel:`Log as Note`.
```

   .. image:: sms_essentials/automated-action-sms.png
      :align: center
      :alt: Automated action template with action to do, SMS template and log as note highlighted.

   :guilabel:`Save` the completed automated action to implement the change. Ensure the

>    :guilabel:`Active` toggle is set to *on* for the automated action to run.

Reviewer 2: I would flip these sentences, so the user *ends* by saving *after* they've hit the 'Active' toggle
*and* since you *DO* mention the toggle here, you can disregard my previous comment/suggestion about it. 👍
